### DISCUSSION TITLE
Aws and fooocus

### DISCUSSION THREAD

**Discussion by rivboyj** [lllyasviel/Fooocus#1386] [Q&A] [answered]
I've been using the Fooocus that I found on the marketplace:https://aws.amazon.com/marketplace/pp/prodview-4xzjmoi4f3ml4

1.MAINLY I would like to know how to import my local models from CIVITai to the aws model

2.Im wondering if anyone else had experience with this. Would love to discuss further. 

3.im just wondering how to get better Full body pictures, it seems to avoid doing this even when asking explicitly. Also I want to get better at making a character, even when i put in FaceSwap it doesn't do that great of a job.

**Comment by tbuyle** [comment]
What you want to do is add the files from CIVITAI to the fooocus/models/checkpoints or fooocus/models/loras

I don't know how to do that with this AWS distribution. You should probably ask meetrix.io since there are also mentionned as support contact on the AWS marketplace.

**Reply by kukovinetsyevhenii** [reply]
That's also what I want to know. I would like to find serverless option for running Foocus on AWS. Sagemaker , Lambda.

**Reply by mashb1t** [reply, marked as answer]
Lambda will never be possible in the way you think it would as Fooocus uses 2 processes (one main GUI thread and one worker), which are countering the approach of the lambda.
Sagemaker is also not the perfect match as Fooocus is not just one model but multiple ones and a whole pipeline.
The best way to use Fooocus in AWS should be by creating a custom AMI and boot up instances on-demand in a 0-0-1 ASG on demand, boot up time ~10min on a Windows machine, ~5min on an Ubuntu server using A10Gs (g5) or T4 (g4dn, not recommended).

**Reply by kukovinetsyevhenii** [reply]
Thank you mashb1t, could you recommend another Stable diffusion open source that can be run on AWS as serverless?
And would like to know which platform is the best choice for Foocus to be deployed in terms of cost and reliability?
Looking forward to your response.

**Reply by mashb1t** [reply]
All of them basically work similarly, so if you want to use SDXL serverless you can use [Sagemaker JumpStart](https://sagemaker-examples.readthedocs.io/en/latest/introduction_to_amazon_algorithms/jumpstart_text_to_image/Amazon_JumpStart_Text_To_Image.html#2.1.-Select-a-Model) or SageMaker Realtime Inference with [SDXL](https://aws.amazon.com/marketplace/pp/prodview-pe7wqwehghdtm).
Price to performance leader strongly depends on how long you want to keep using the same instance type, as you can save up to 70% with a 3 year savings plan (not recommended due to a frequently changing AI landscape). Cheaper alternatives are GCP (instances less powerful depending on region) or HuggingFace, but you also might be better off with using Fooocus in https://rundiffusion.com/ (paid by the hour or by tokens).

**Comment by kukovinetsyevhenii** [comment]
Thank you. I understand
I would like to install Foocus on Hosting service, don't want to use external API that 's already published.

**Comment by jeanjerome** [comment]
And what about Fooocus on AWS EKS? Has anyone deployed Fooocus on K8s, and what would be your advice or recommendations for doing so? Thank you.

**Reply by mashb1t** [reply]
@jeanjerome Here is an example for EKS for deep learning, but i assume it's similar for Fooocus and inference in general: https://docs.aws.amazon.com/deep-learning-containers/latest/devguide/deep-learning-containers-eks-setup.html

You might also find ComfyUI examples, which you coult then adapt.

**Reply by jeanjerome** [reply]
Thanks @mashb1t, I’ll take a look.

**Reply by MaximeMRF** [reply]
Hey @jeanjerome I made a [repository](https://github.com/trackit/eks-auto-mode-gpu) to deploy Fooocus on AWS EKS, feel free to check the project

**Reply by jeanjerome** [reply]
Hi @MaximeMRF, superb implementation, everything looks very clean. Thank you.
